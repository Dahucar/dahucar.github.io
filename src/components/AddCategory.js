import React, { useState } from 'react';
import PropTypes from 'prop-types'

// el metodo enviado desde otro componente se recive en los props
const AddCategory = ({ setCategories }) => {
  const [ inputValue, setInputValue ] = useState('');
  const handleChangeInputValue = ( e ) => {
    setInputValue( e.target.value );
  }

  const handleSubmitForm = ( e ) => {
    console.log('handleSubmitForm', inputValue);
    e.preventDefault();
    // a qui se llama al metodo que se recive desde los props en el por referencia se tendra el listado de categorias.

    if (inputValue.trim().length >= 2) {
      setCategories( categ =>  [ inputValue, ...categ ] );
      setInputValue( '' );
    }

  } 

  return (
    <form onSubmit={ handleSubmitForm }>
      <input 
        type="text"
        value={ inputValue }
        placeholder="Ingrese un nombre"
        onChange={ handleChangeInputValue }
      />
      <span>{ inputValue }</span>
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;