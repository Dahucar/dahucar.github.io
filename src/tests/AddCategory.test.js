import React from 'react'
import { shallow } from 'enzyme';
import AddCategory from '../components/AddCategory';

describe('Testing on AddCategory Component', () => {
  // with jest.fn function, we will know if have change in the state
  const setCategories = jest.fn();
  let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
  
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
  });
  
  test('the AddCategory component should be show with not problem', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should be changue the input value', () => {
    //Search the input on the html content
    const input = wrapper.find('input');
    const inputValue = 'I´m a message for testing input change event';
    
    //simulate a input change event 
    input.simulate('change', {target: { value: inputValue }});
    expect( wrapper.find('span').text().trim() ).toBe( inputValue );
  });
  
  test('Information should not be displayed in the console', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect( setCategories ).not.toHaveBeenCalled();
  });

  test('Should be call setCategories function and after clear the input value', () => {
    // Activities:
    //  -> simulate the input change event
    const inputForm = wrapper.find('input');
    inputForm.simulate('change', {target: { value: 'I´m a test value' }});
    //  -> simulate the submit event
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    //  -> call the setCategories function
    expect( setCategories ).toHaveBeenCalled();
    expect( setCategories ).toHaveBeenCalledTimes(1);
    expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );
    //  -> the input value should be empty
    expect( inputForm.prop('value') ).toBe( '' );
  });
  
});
