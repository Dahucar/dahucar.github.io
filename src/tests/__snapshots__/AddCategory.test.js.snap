// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on AddCategory Component the AddCategory component should be show with not problem 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Ingrese un nombre"
    type="text"
    value=""
  />
  <span />
</form>
`;
